taipy/core/taipy.py

```diff
@@ -973,6 +998,106 @@ def export_scenario(
+
+    if not ((folder / "scenarios").exists() or (folder / "scenario").exists()):
+        raise ImportFolderDoesntContainAnyScenario(folder_path)
+
+    if not (folder / "version").exists():
+        raise ImportScenarioDoesntHaveAVersion(folder_path)
+
+    entity_managers = {
+        "cycles": _CycleManagerFactory._build_manager,
+        "cycle": _CycleManagerFactory._build_manager,
+        "data_nodes": _DataManagerFactory._build_manager,
+        "data_node": _DataManagerFactory._build_manager,
+        "tasks": _TaskManagerFactory._build_manager,
+        "task": _TaskManagerFactory._build_manager,
+        "scenarios": _ScenarioManagerFactory._build_manager,
+        "scenario": _ScenarioManagerFactory._build_manager,
+        "jobs": _JobManagerFactory._build_manager,
+        "job": _JobManagerFactory._build_manager,
+        "submission": _SubmissionManagerFactory._build_manager,
+        "version": _VersionManagerFactory._build_manager,
+    }
+
+    entity_managers["version"]()._import(next((folder / "version").iterdir()), "")
+
+    valid_entity_folders = list(entity_managers.keys())
+    valid_data_folder = Config.core.storage_folder
+
+    imported_scenario = None
+    imported_entities = []
+
+    for entity_folder in folder.iterdir():
+        if not entity_folder.is_dir() or entity_folder.name not in valid_entity_folders + [valid_data_folder]:
+            __logger.warning(f"{entity_folder} is not a valid Taipy folder and will not be imported.")
+            continue
+
+    for entity_type in valid_entity_folders:
+        # Skip the version folder as it is already handled
+        if entity_type == "version":
+            continue
+
+        entity_folder = folder / entity_type
+        if not entity_folder.exists():
+            continue
+
+        manager = entity_managers[entity_type]()
+        for entity_file in entity_folder.iterdir():
+            # Check if the to-be-imported entity already exists
+            entity_id = entity_file.stem
+            if manager._exists(entity_id):
+                if override:
+                    __logger.warning(f"{entity_id} already exists and will be overridden.")
+                else:
+                    __logger.error(f"{entity_id} already exists. Please use the 'override' parameter to override it.")
+                    raise EntitiesToBeImportAlredyExist(folder_path)
+
+            # Import the entity
+            imported_entity = manager._import(
+                entity_file,
+                version=_VersionManagerFactory._build_manager()._get_latest_version(),
+                data_folder=folder / valid_data_folder,
+            )
+
+            imported_entities.append(imported_entity)
+            if entity_type in ["scenario", "scenarios"]:
+                imported_scenario = imported_entity
+
+    __logger.info(f"Scenario {imported_scenario.id} has been successfully imported.")  # type: ignore[union-attr]
+    return imported_scenario
```

Comment:
This function is complex. It seems too complex to remain in the taipy.py file, which is the top-level API. 
Would delegating most of this logic to the manager classes make sense?

Reply:
I agree. I will try to refactor this next